js/src/main/scala/scalajson/ast/JValue.scala

```diff
@@ -77,22 +77,26 @@ object JNumber {
   def apply(value: Float): JValue = value match {
     case n if java.lang.Float.isNaN(n) => JNull
     case n if n.isInfinity => JNull
-    case _ => JNumber(value.toString)
+    case _ => new JNumber(value.toString)
   }
+
+  def fromString(value: String): Option[JNumber] =
+    if (value.matches(jNumberRegex))
+      Some(new JNumber(value))
+    else
+      None
+
+  def unapply(arg: JNumber): Option[String] = Some(arg.underlying)
 }
 
 /** Represents a JSON number value. If you are passing in a
-  * NaN or Infinity as a [[scala.Double]], [[JNumber]] will
+  * NaN or Infinity as a [[scala.Double]] or [[scala.Float]], [[JNumber]] will
   * return a [[JNull]].
   *
   * @author Matthew de Detrich
-  * @throws scala.NumberFormatException - If the value is not a valid JSON Number
   */
-final case class JNumber(value: String) extends JValue {
-
-  if (!value.matches(jNumberRegex)) {
-    throw new NumberFormatException(value)
-  }
+final class JNumber(private[ast] val underlying: String) extends JValue {
+  @inline val value: String = underlying
```

Comment:
> To access the raw string value (you can't access value because it's private and for large numbers you may want to access the String value).

I don't see the benefit of adding a second `value` member. 
Why not simply call the constructor parameter `value` and make it a public field?